### PR TITLE
chore: Improve js execution instrumentation

### LINKED ISSUE #24706: [Task]: Better instrumentation for JS execution
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

1. Modify the payload for EXECUTE_ACTION event on JSObject function execution
```
{
    "type": "JS",  
    "name": "myFun1",  // to be changed. only send the function name.
    "JSObjectName": "JSObject1",   // to be added
    "pageId": "62f0b9e9b2f8027edfbf9008",   // to be added
    "appId": "62f0b9e9b2f8027edfbf9005",   // to be added
    "appMode": "EDIT",         // to be added
    "appName": "Testbed1123",  // to be added
    "isExampleApp": false,  // to be added
    "actionId": "62f4a31274d6e95d0a544da8",      // to be added
    "userData": {
        "userId": "",
        "email": "",
        "appId": "",
        "source": "cloud"
    },
    "instanceId": ""  // to be added
}
```

2. Track JS Expression execution on event handlers
```
{
    "type": "JS_EXPRESSION",  
    "unevalValue": "{{console.log("hello")}}" 
    "pageId": "62f0b9e9b2f8027edfbf9008",   
    "appId": "62f0b9e9b2f8027edfbf9005",   
    "appMode": "EDIT",         
    "appName": "Testbed1123",  
    "isExampleApp": false,  
    "actionId": "62f4a31274d6e95d0a544da8", 
    "userData": {
        "userId": "",
        "email": "",
        "appId": "",
        "source": "cloud"
    },
    "instanceId": ""  
}
```

3. Track execution success and failures for JSObject functions and JS Expressions
`EXECUTE_ACTION_SUCCESS` or `EXECUTE_ACTION_FAILURE`
```
{
    "type": "JS",  
    "name": "myFun1", 
    "JSObjectName": "JSObject1",  
    "pageId": "62f0b9e9b2f8027edfbf9008",   
    "appId": "62f0b9e9b2f8027edfbf9005",   
    "appMode": "EDIT",         
    "appName": "Testbed1123",  
    "isExampleApp": false,  
    "actionId": "62f4a31274d6e95d0a544da8", 
    "userData": {
        "userId": "",
        "email": "",
        "appId": "",
        "source": "cloud"
    },
    "instanceId": ""  
}
```

### PR DESCRIPTION
**Pull request by ohansFavour**
###Description

This PR enriches the data logged for the EXECUTE_ACTION event on trigger fields.

Example schema
Screenshot 2023-07-05 at 10 03 32

Fixes #24706 

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by Rishabh-Rathod — DISMISSED** [review]

File: app/client/src/workers/Evaluation/fns/utils/TriggerEmitter.ts [hunk 1]
```diff
@@ -182,15 +186,21 @@ TriggerEmitter.on(
...
-export const fnInvokeLogHandler = priorityBatchedActionHandler<string>(
-  (data) => {
-    const set = new Set([...data]);
-    WorkerMessenger.ping({
-      method: MAIN_THREAD_ACTION.LOG_JS_FUNCTION_EXECUTION,
-      data: [...set],
-    });
-  },
-);
+export const fnInvokeLogHandler = priorityBatchedActionHandler<{
```
Comment (Rishabh-Rathod) [inline app/client/src/workers/Evaluation/fns/utils/TriggerEmitter.ts:189]: Shall we use `deferredBatchedActionHandler` for the logs ?

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM.

Only a small nitpick

File: app/client/src/workers/Evaluation/fns/utils/TriggerEmitter.ts (hunk 1 above)
Comment (ohansFavour) [reply app/client/src/workers/Evaluation/fns/utils/TriggerEmitter.ts:189]: This makes sense

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

**Pull request merged**